File: Kotlin/Lab8/src/by/bntu/fitr/povt/cng/model/util/logging/FileLogger.java
package by.bntu.fitr.povt.cng.model.util.logging;

import by.bntu.fitr.povt.cng.model.util.logging.Decorator.TextDecorator;

import java.io.*;

public class FileLogger extends AbstractLogger implements Logger {

    private String fileName;
    private TextDecorator decorator;

    public FileLogger() {
        fileName = "fileLogger.txt";
    }

    public FileLogger(String fileName, TextDecorator decorator) {
        this.fileName = fileName;
        this.decorator = decorator;
    }

    public FileLogger(String fileName) {
        this.fileName = fileName;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public void logEvent(String string) {
        var baseString = super.getBaseString();
        try (var writer = new FileWriter(new File(fileName), true)){
            writer.append(baseString).append(string).append("\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: Kotlin/TryKotlin/src/by/bntu/fitr/povt/cng/model/loggers/FileLogger.kt
package by.bntu.fitr.povt.cng.model.loggers

import by.bntu.fitr.povt.cng.model.decorators.CaseDecorator
import java.io.File

class FileLogger : AbstractLogger, Logger {

    private var decorator: CaseDecorator? = null
    private var pathToFile: String? = null

    constructor()

    constructor(file: String) : super() {
        this.pathToFile = file
    }

    constructor(decorator: CaseDecorator, pathToFile: String) {
        this.decorator = decorator
        this.pathToFile = pathToFile
    }


    override fun logEvent(event: String) {
        if (decorator != null) {
            File(pathToFile).printWriter().use { out ->
                {
                    out.println(decorator!!.decorate("${super.getDate()}:\t $event"))
                }
            }
        } else {
            File(pathToFile).printWriter().use { out ->
                {
                    out.println("${super.getDate()}:\t $event")
                }

            }
        }

    }


}